sdk/storage/azblob/chunkwriting.go

```diff
@@ -33,7 +33,7 @@ type blockWriter interface {
 // well, 4 MiB or 8 MiB, and auto-scale to as many goroutines within the memory limit. This gives a single dial to tweak and we can
 // choose a max value for the memory setting based on internal transfers within Azure (which will give us the maximum throughput model).
 // We can even provide a utility to dial this number in for customer networks to optimize their copies.
-func copyFromReader(ctx context.Context, from io.ReadSeekCloser, to blockWriter, o UploadStreamToBlockBlobOptions) (BlockBlobCommitBlockListResponse, error) {
+func copyFromReader(ctx context.Context, from io.Reader, to blockWriter, o UploadStreamToBlockBlobOptions) (BlockBlobCommitBlockListResponse, error) {
```

Comment:
We're moving to use io.ReadSeekCloser across all the SDKs. You can write simple struct like [here](https://github.com/Azure/azure-sdk-for-go/blob/dfdc424c554cfb6e45a53e259c5c215bb4e1d3a6/sdk/storage/azblob/internal/zc_shared.go#L27)

Reply:
Right. I'd implemented that in my PR to workaround for now, but then if this interface does change to use these unused methods that'll introduce a bug rather than a compile time error. It also makes the interface more confusing to use because then I can't tell from the function signature whether the reader will be kept open or not

But alright, if there's some design ideology at play I'll have to accept it

Reply:
For context, we require a seekable stream to facilitate our retry logic.  If the PUT request fails, we can rewind the stream to try again.

@mohsha-msft I'm confused in this case.  The API takes a `ReadSeekCloser` but doesn't actually require a `Seeker` or a `Closer`.  Is it used elsewhere (maybe via a type-assertion)?

Reply:
@serprex sorry, just to clarify, this is a private method right? How are you leveraging it exactly?

Reply:
It's called from `BlockBlobClient.UploadStreamToBlockBlob()` which also takes a `ReadSeekCloser`

Reply:
I think this requirement is hard to meet in services.
Often I stream data right from a client through my service to blob storage. The HTTP connection I am feeding from can't be a `Seeker`. If azblob wants a retry mechanism, I think it should implement this internally with some kind of (configurable?) buffer. The burden should not be on the caller.